METAPROGRAMMING.md

```diff
@@ -1,37 +1,104 @@
 # Metaprogramming with decorators
 
 Previously, metaprogramming within JavaScript was driven by use of dynamic operations that manipulate JavaScript objects, such as `Object.defineProperty`. ESnext upgrades classes with decorators, with the advantages:
-- Decorators make it more clear what's going on--users write class literals and explicitly annotate the piece of source code that is being manipulated, rather than spooky action-at-a-distance where a big object is passed into a special application-dependent class framework.
-- Decorators can manipulate private fields and private methods, whereas operations on objects are unable to manipulate the definitions of private names.
-- Decorators do their manipulations at the time the class is defined, rather than when instances are being created and used, so they may lead to patterns which are more amenable to efficient implementation.
 
-Decorators are implemented as functions which take a JSON-like representation of a class element as an argument and return a possibly-modified form of that, optionally with additional class elements. Class elements are represented in the form: 
+* Users write class literals and explicitly annotate the piece of source code that is being manipulated, rather than spooky action-at-a-distance where a big object is passed into a special application-dependent class framework.
+* Decorators can manipulate private fields and private methods, whereas operations on objects are unable to manipulate the definitions of private names.
+* Decorators do their manipulations at the time the class is defined, rather than when instances are being created and used, so they may lead to patterns which are more amenable to efficient implementation.
+
+Dcorators can be used either to decorate a whole class or an individual class element (field or method). Decorators are implemented as functions which take a JSON-like representation of class element(s) as an argument and return a possibly-modified form of that, optionally with additional class elements.
+
+The signatures of the different kinds of decorator functions are as follows:
+
+### Field Decorator
```

Comment:
Could you put an example of usage here, and then explain that the decorator is a function which is called with the descriptor as an argument?